Kernel/Devices/SerialDevice.cpp

```diff
@@ -140,4 +100,175 @@ u8 SerialDevice::get_line_status() const
+        // not -ENOTIMPL, since it's a hardware limitation
+        return -EINVAL;
+    }
+    switch (in_baud) {
+    case B50:
+        set_baud(Baud::Baud50);
+        break;
+    case B110:
+        set_baud(Baud::Baud110);
+        break;
+    case B300:
+        set_baud(Baud::Baud300);
+        break;
+    case B600:
+        set_baud(Baud::Baud600);
+        break;
+    case B1200:
+        set_baud(Baud::Baud1200);
+        break;
+    case B2400:
+        set_baud(Baud::Baud2400);
+        break;
+    case B4800:
+        set_baud(Baud::Baud4800);
+        break;
+    case B9600:
+        set_baud(Baud::Baud9600);
+        break;
+    case B19200:
+        set_baud(Baud::Baud19200);
+        break;
+    case B38400:
+        set_baud(Baud::Baud38400);
+        break;
+    case B57600:
+        set_baud(Baud::Baud57600);
+        break;
+    case B115200:
+        set_baud(Baud::Baud115200);
+        break;
+    default:
+        dbgln("{}: Tried to set unsupported baud speed", m_tty_name);
+        return -EINVAL;
+    }
+    return 0;
+}
+
+int SerialDevice::change_parity(TTY::ParityMode parity_mode)
+{
+    switch (parity_mode) {
+    case TTY::ParityMode::None:
+        set_line_control(ParitySelect::None, m_stop_bits, m_word_length);
+        break;
+    case TTY::ParityMode::Odd:
+        set_line_control(ParitySelect::Odd, m_stop_bits, m_word_length);
+        break;
+    case TTY::ParityMode::Even:
+        set_line_control(ParitySelect::Even, m_stop_bits, m_word_length);
+        break;
+    }
+    return 0;
+}
+
+int SerialDevice::change_stop_bits(TTY::StopBits stop_bits)
+{
+    switch (stop_bits) {
+    case TTY::StopBits::One:
+        set_line_control(m_parity_select, StopBits::One, m_word_length);
+        break;
+    case TTY::StopBits::Two:
+        set_line_control(m_parity_select, StopBits::Two, m_word_length);
+        break;
+    }
+    return 0;
+}
+
+int SerialDevice::change_character_size(TTY::CharacterSize character_size)
+{
+    switch (character_size) {
+    case TTY::CharacterSize::FiveBits:
+        set_line_control(m_parity_select, m_stop_bits, WordLength::FiveBits);
+        break;
+    case TTY::CharacterSize::SixBits:
+        set_line_control(m_parity_select, m_stop_bits, WordLength::SixBits);
+        break;
+    case TTY::CharacterSize::SevenBits:
+        set_line_control(m_parity_select, m_stop_bits, WordLength::SevenBits);
+        break;
+    case TTY::CharacterSize::EightBits:
+        set_line_control(m_parity_select, m_stop_bits, WordLength::EightBits);
+        break;
+    }
+    return 0;
+}
+
+int SerialDevice::change_receiver_enabled(bool should_enable)
+{
+    if (!should_enable) {
+        dbgln("{}: FIXME: we don't have a way to disable serial yet", m_tty_name);
+        return -ENOTIMPL;
+    }
+    return 0;
+}
+
+int SerialDevice::change_ignore_status_lines(bool should_enable)
+{
+    m_ignore_status_lines = should_enable;
```

Comment:
this is the only place in which the member is accessed, what does this do?

Reply:
This will be needed when error (parity bit fail, buffer overrun) handling will be implemented.